### PR TITLE
Add Tests Around Load Image.

### LINKED ISSUE #10451: Add more complex integration tests for `minikube image load`
In #10366  I added a simple test for the `minikube image load` command:
1. Pull busybox
2. Try to load it into minikube
3. Make sure busybox exists in minikube

It would be nice to beef up the tests a little bit and also verify the following cases:
1. Try to load an image that already is loaded in, and make sure nothing breaks
2. Load in an image with a new tag and make sure the new tag exists in the daemon

If anyone would be interested in picking this up, please feel free to assign this issue to yourself by commenting /assign.

### PR DESCRIPTION
**Pull request by vishjain**
fixes #10451

### CODE REVIEW

File: test/integration/functional_test.go [hunk 1]
```diff
@@ -230,21 +252,42 @@ func validateLoadImage(ctx context.Context, t *testing.T, profile string) {
...
+func validateLoadImageAndUpdateTag(ctx context.Context, t *testing.T, profile string) {
+	if NoneDriver() {
+		t.Skip("load image not available on none driver")
+	}
+	if GithubActionRunner() && runtime.GOOS == "darwin" {
+		t.Skip("skipping on github actions and darwin, as this test requires a running docker daemon")
+	}
+	defer PostMortemLogs(t, profile)
+	// pull busybox
+	busyboxImage := "busybox:1.33"
```
Comment (medyagh) [inline test/integration/functional_test.go:269]: lets use a GCR hosted image to avoid the rate limit

File: test/integration/functional_test.go [hunk 2]
```diff
@@ -207,6 +208,27 @@ func validateNodeLabels(ctx context.Context, t *testing.T, profile string) {
...
+	if err != nil {
+		t.Fatalf("loading image into minikube: %v\n%s", err, rr.Output())
+	}
+
+	// make sure the image was correctly loaded
+	rr, err = inspectImage(ctx, t, profile, newImage)
+	if err != nil {
+		t.Fatalf("listing images: %v\n%s", err, rr.Output())
+	}
+	if !strings.Contains(rr.Output(), fmt.Sprintf("busybox:load-%s", profile)) {
```
Comment (tharun208) [inline test/integration/functional_test.go:227]: can we do the comparison against the `newImage` argument?

File: test/integration/functional_test.go [hunk 3]
```diff
@@ -230,21 +252,42 @@ func validateLoadImage(ctx context.Context, t *testing.T, profile string) {
...
+// validateLoadImageAndUpdateTag loads image inspects, then updates tag and checks correctness.
+func validateLoadImageAndUpdateTag(ctx context.Context, t *testing.T, profile string) {
+	if NoneDriver() {
+		t.Skip("load image not available on none driver")
+	}
+	if GithubActionRunner() && runtime.GOOS == "darwin" {
+		t.Skip("skipping on github actions and darwin, as this test requires a running docker daemon")
+	}
+	defer PostMortemLogs(t, profile)
+	// pull busybox
```
Comment (tharun208) [inline test/integration/functional_test.go:268]: please modify the comments

**Review by medyagh — CHANGES_REQUESTED** [review]
would u plz update the test comment with with more information ?

both for the new test and also the preivous test
"validateLoadImage"


by reading the doc we should like to learn what these two tests two differently and we why we have two diff tests and not combined into one?

**Comment by vishjain** [comment]
Closing this and putting a new PR with comments. @medyagh

**Pull request closed without merging**